tools/perf_analysis/plot_pointwise_data.py

```diff
@@ -0,0 +1,132 @@
+#                 plot_pointwise_data.py
+#                 Author:  Brian Pickrell, AMD copyright 2022
+
+#   ingest the JSON files created by  test_broadcast_perf.py
+#   and print optimum local/global inputs for the test set.
+#   Use this code as an example to make pyplot charts or
+#   otherwise do follow-up analysis of the data.
+#
+#   syntax:  python3 plot_pointwise_data.py <input JSON file>
+
+from matplotlib import pyplot as plt
+import json
+import os
+import sys
+import numpy as np
+import pandas
+
+if len(sys.argv) < 2:
+    print('syntax: python3 ' + __file__ +
+          " .tools/perf_analysis/pointwise/pointwise_perf_results_807270.json")
+    exit(-1)
+
+with open(sys.argv[1], "r") as read_file:
+    data = json.load(read_file)
+
+# Find the directory of the input file; output will go there
+output_dir = os.path.abspath(os.path.dirname(sys.argv[1]))
+
+# Example:  to view the keys of the top level JSON nodes,
+# print('keys are ', data.keys())
+
+# Example:  to get a JSON subnode,
+# tensors = data['tensors']
+# print('length of tensors=', len(tensors))
+
+################################################################################################################
+#
+#         Demonstration of loading and filtering the data in tabular form.  Column headers are hardcoded here.
+#
+################################################################################################################
+
+result = data["result"]
+df = pandas.DataFrame(result,
+                      columns=[
+                          'tensorsize', 'global_workitems',
+                          'local_workitems_per_CU', 'time'
+                      ])
+
+# print(df)
+
+
+# Given three dataframe columns (column labels), plot one as a function of the other two.
+#
+# the_df  pandas.DataFrame
+# param_a string is the dependent variable.  (column label)
+# param_b string is the categorical variable, i.e. there is one stacked line for each unique value of b (column label)
+# param_c string  is the independent variable (the x axis) (column label)
+#
+def plot_vs_parameter(the_df, param_a, param_b, param_c):
+    the_df = the_df.sort_values(by=[param_b, param_c])
+    # make a list of unique global values
+
+    # issue: we get a different range of param_b (global) for different param_a (tensor size)
+    # each tensorsize should get its own legend??
+    ss = np.unique(the_df[param_b])
+    # print('ss=',ss)
```

Comment:
is this comment needed?

Reply:
Removed.  I've left several other print statements in comment form where they show syntax for list indexing and slicing that's not self-evident.